File: src/app/post.ts
/**
 * holds data for a post
 */
export class Post {
  constructor(id?: number) {
    this.id = id;
  }
  username: string;
  name: string;
  body: string;
  likes: number;
  comments: number;
  userId: number;
  id: number;
}

File: src/main/java/com/app/woofer/model/Post.java
package com.app.woofer.model;

import com.app.woofer.model.User;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.Instant;

@Entity
@Table(name = "posts")
@Data
@NoArgsConstructor
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private int id;

    @ManyToOne
    @JoinColumn(name = "userID", referencedColumnName = "id")
    private User user;

    private Instant timestamp;
    @Column(insertable = false, updatable = false)
    private int userID; //delete this?
    private String body;
}
